jenkins-webhook/1.0.0/README.md

```diff
@@ -0,0 +1,120 @@
+
+1. Store the Keptn API token as a Jenkins credential
+2. Parameterise the Jenkins build
+3. Trigger the job using `/buildWithParameters` endpoint
+4. On triggering a job, send important Keptn parameters to Jenkins as Job parameters (see below)
+5. (optional) Use build parameters sent above to send `taskName.status.changed` events during execution
+6. Use build parameters sent above to craft a `taskName.finished` event and send to Keptn
+
+### Jenkins Build Parameters
+Parameterise the Jenkins job and add the following parameters:
+
+1. `keptnProject`, `keptnService` and `keptnStage`
+2. `keptnContext` and `keptnId`
+3. `keptnTask`
+
+### Craft Webhook Requests
+
+The URL should be a POST and look something like this:
+
+```
+https://myJenkins.com/job/JobName/buildWithParameters?keptnProject={{.data.project}}&keptnService={{.data.service}}&keptnStage={{.data.stage}}&keptnContext={{.shkeptncontext}}&keptnId={{.id}}&keptnTask=<YourTaskName>
+```
+
+![sync webhook](https://raw.githubusercontent.com/keptn-contrib/artifacthub/main/jenkins-webhook/1.0.0/assets/sync-webhook.jpg)
+
+Items in double curly brackets are variables which will be pulled from the `taskName.triggered` CloudEvent and replaced at runtime.
+
+These parameters are required so you can build a `taskName.finished` event when the job has finished.
+
+Sending one or more `taskName.status.changed` event(s) **during** execution is also possible. Adjust the example below so the `type` is `sh.keptn.event.<taskName>.status.changed` rather than `sh.keptn.event.<taskName>.finished`.
+
+### Sample Pipeline
+
+```
+pipeline {
+    agent any
+
+    stages {
+        stage('Hello') {
+            steps {
+                
+                echo 'Hello World'
+                sleep 10
+                
+                withCredentials([string(credentialsId: 'keptn-api-token', variable: 'KEPTN_API_TOKEN')]) {
+                  sh """
+                    curl -X POST http://myKeptn.com/api/v1/event \
+                    -H 'x-token: ${KEPTN_API_TOKEN}' \
+                    -H 'Content-Type: application/json' \
+                    --data-raw '{
+                                "specversion": "1.0",
+                                "source": "Jenkins",
+                                "shkeptncontext": "${params.keptnContext}",
+                                "triggeredid": "${params.keptnId}",
+                                "type": "sh.keptn.event.${params.keptnTask}.finished",
+                                "contenttype": "application/json",
+                                "data": {
+                                           "project": "${params.keptnProject}",
+                                           "service": "${params.keptnService}",
+                                           "stage": "${params.keptnStage}",
+                                           "status": "succeeded",
+                                           "result": "pass"
```

Comment:
Can we point out that we can also send other statuses and results according to the [Keptn spec](https://github.com/keptn/spec/blob/master/cloudevents.md#data)?
> status: indicates whether the service executing the task was able to perform the task without any unexpected errors. Possible values are succeeded, errored,or unknown
> result: indicates the result of a successful task execution without unexpected problems (i.e. status = succeeded), such as the result of an evaluation, or a test execution. Possible values are pass, warning, or fail

I think this is really a handy way to signal Keptn that your Jenkins pipeline failed.